src/vmm/src/vstate/vm.rs

```diff
@@ -278,6 +286,47 @@ impl Vm {
     }
 }
 
+/// Use `mincore(2)` to overapproximate the dirty bitmap for the given memslot. To be used
+/// if a diff snapshot is requested, but dirty page tracking wasn't enabled.
+fn mincore_bitmap(region: &GuestRegionMmap) -> Result<Vec<u64>, VmError> {
+    // TODO: Once Host 5.10 goes out of support, we can make this more robust and work on
+    // swap-enabled systems, by doing mlock2(MLOCK_ONFAULT)/munlock() in this function (to
+    // force swapped-out pages to get paged in, so that mincore will consider them incore).
+    // However, on AMD (m6a/m7a) 5.10, doing so introduces a 100%/30ms regression to snapshot
+    // creation, even if swap is disabled, so currently it cannot be done.
```

Comment:
nit: Should we create a GitHub issue to track this TODO as the PR checklist says?